macOS/DuckDuckGo/Application/AppDelegate.swift

```diff
@@ -291,6 +291,18 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
                                                                    canPerformAuthMigration: true,
                                                                    canHandlePixels: true)
 
+            // Expired refresh token recovery
+            if #available(iOS 15.0, macOS 12.0, *) {
+                let restoreFlow = DefaultAppStoreRestoreFlowV2(subscriptionManager: subscriptionManager, storePurchaseManager: subscriptionManager.storePurchaseManager())
+                subscriptionManager.tokenRecoveryHandler = {
+                    try await DeadTokenRecoverer.attemptRecoveryFromPastPurchase(subscriptionManager: subscriptionManager, restoreFlow: restoreFlow)
+                }
+            } else {
+                subscriptionManager.tokenRecoveryHandler = {
+                    try await DeadTokenRecoverer.attemptRecoveryFromPastPurchase(subscriptionManager: subscriptionManager)
```

Comment:
Do we need this call here, for the non supported OS (which is now only macOS under 12.0) this will be a no-op.

Reply:
the call is just for logging the issue and sending the notification, nothing too bad